src/Nix/Builtins.hs

```diff
@@ -828,7 +836,7 @@ fromJSON = fromValue >=> \encoded ->
 
 toXML_ :: MonadNix e m => m (NValue m) -> m (NValue m)
 toXML_ v = v >>= normalForm >>= \x ->
-    pure $ nvStr (Text.pack (toXML x)) mempty
+    pure $ nvStr $ makeNixStringWithoutContext $ Text.pack (toXML x)
```

Comment:
`toXML` should output a `NixString`, it may have some context depending on the value in question.